packages/components/src/components/post-megadropdown/post-megadropdown.tsx

```diff
@@ -1,25 +1,33 @@
 import { getFocusableChildren } from '@/utils/get-focusable-children';
 import { Component, Element, Event, EventEmitter, h, Host, Method, State } from '@stencil/core';
-import { DEVICE_SIZE } from '../post-header/post-header';
+import { breakpoint } from '../../utils/breakpoints';
 
 @Component({
   tag: 'post-megadropdown',
   styleUrl: 'post-megadropdown.scss',
   shadow: false,
 })
 export class PostMegadropdown {
-  private header: HTMLPostHeaderElement | null;
-
   private firstFocusableEl: HTMLElement | null;
   private lastFocusableEl: HTMLElement | null;
 
-  @State() device: DEVICE_SIZE;
+  @State() device: string = breakpoint.get('name');
 
   @Element() host: HTMLPostMegadropdownElement;
 
   /** Tracks the currently active dropdown instance. */
   private static activeDropdown: PostMegadropdown | null = null;
 
+  private breakpointChange(e: CustomEvent) {
+    const newDevice = e.detail; 
+    if (this.device !== newDevice) { 
```

Comment:
Is this check even necessary? I think the postBreakpoint:name event should only trigger, if there really is a change. Can you test that and remove the line here, if it's unnecessary?

And also to save an object property in a new var/const is unnecessary. Instead of `const newDevice = e.detail` and then using `newDevice`, you can just use `e.detail` and save some Bits in memory.